### PR TITLE
Add script to walk through a directory tree and split all the routine…

### PR DESCRIPTION
**Pull request by bimaoe**
… charts.

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -354,7 +354,7 @@ def split_routine(ssc_file, target_description, result_file, chart_type):
 			break
 
 		# Read multiple line configs.
-		while line[-2] != ';':
+		while line.split()[0] != ';':
```
Comment (bimaoe) [inline utils.py]: Undo

File: walk.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import os
+import re
+import subprocess
+import shutil
+import sys
+from utils import *
+
+if len(sys.argv) < 2:
```
Comment (zjma) [inline walk.py]: <=2

File: walk.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+from utils import *
+
+if len(sys.argv) < 2:
+	print('Usage: walk.py <root_directory> <target_root_directory>')
+	exit()
+
+root_dir = sys.argv[1]
+target_root_dir = sys.argv[2]
+
+print('Splitting all routine charts inside "{}".'.format(root_dir))
```
Comment (zjma) [inline walk.py]: nit:
```
print(f'Splitting all routine charts inside "{root_dir}".')
```

File: walk.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import os
```
Comment (zjma) [inline walk.py:1]: it so far only works in Windows. Let's note it somewhere.

File: walk.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+# Walk on all subdirectories and find the ones that have sscs.
+for sub_dir, dirs, files in os.walk(root_dir):
+	for filename in files:
+		# If the directory contains an ssc, check if it has a rountine chart.
+		if filename.endswith('.ssc'):
+			filepath = sub_dir + os.sep + filename
+			print('Processing {0}'.format(filepath))
+			# Find and split all the routine charts.
+			subprocess.call(['python', 'process.py', filepath, 'tmp'])
```
Comment (zjma) [inline walk.py]: This assumes the working directory is where `process.py` lives, which may not.

File: walk.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import shutil
+import sys
+from utils import *
+
+if os.name != 'nt':
+	print('As of now, only Windows is supported. :(')
+	exit()
+
+if len(sys.argv) < 3:
+	print('Usage: walk.py <root_directory> <target_root_directory>')
```
Comment (zjma) [inline walk.py:12]: NIT: letting user specify a target root is very troublesome. That means you take the responsibility of checking target exists or not, permission stuff... I would suggest to output everything into a $TEMP folder, output the folder path, and let user take care of the rest.

File: walk.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+		# If the directory contains an ssc, check if it has a rountine chart.
+		if filename.endswith('.ssc'):
+			filepath = os.path.join(sub_dir, filename)
+			print(f'Processing {filepath}')
+			# Find and split all the routine charts.
+			subprocess.call(['python', os.path.join(script_path, 'process.py'),
+					filepath, 'tmp'])
+			# Copy all the contents from the source directory to the target.
+			target_sub_dir = sub_dir.replace(root_dir, target_root_dir, 1)
+			shutil.copytree(sub_dir, target_sub_dir)
```
Comment (zjma) [inline walk.py]: If we have `aa/bb/cc.ssc` processed before `aa/dd.ssc`, we will lose the `cc.ssc` we want.

Comment (zjma) [reply walk.py]: how about:
```
for file in all_files:
    if file.endswith('.ssc'):
        #generate new ssc in target location
    else:
        #force copy the file: will create containinig path if not exists
```

**Review by zjma — APPROVED** [review]

**Pull request merged**